setup.py

```diff
@@ -1,49 +1,11 @@
-#!/usr/bin/env python
-
-from setuptools import setup, Extension
-from setuptools.command.test import test as TestCommand
-from setuptools.command.sdist import sdist
-from setuptools.command.build_ext import build_ext
-
-from distutils.errors import DistutilsError, CCompilerError, DistutilsExecError, DistutilsPlatformError
-
+import ast
 import os
 import subprocess
-import sys
-import ast
-
 
-# A py.test test command
-class PyTest(TestCommand):
-    user_options = [('pytest-args=', 'a', "Arguments to pass to py.test"),
-                    ('coverage', 'c', "Generate coverage report")]
-
-    def initialize_options(self):
-        TestCommand.initialize_options(self)
-        self.pytest_args = ''
-        self.coverage = False
-
-    def finalize_options(self):
-        TestCommand.finalize_options(self)
-
-        # The following is required for setuptools<18.4
-        try:
-            self.test_args = []
-        except AttributeError:
-            # fails on setuptools>=18.4
-            pass
-        self.test_suite = 'unused'
-
-    def run_tests(self):
-        import pytest
-        test_args = ['test']
-        if self.pytest_args:
-            test_args += self.pytest_args.split()
-        if self.coverage:
-            test_args += ['--cov', os.path.abspath('asv')]
-        errno = pytest.main(test_args)
-        if errno != 0:
-            raise DistutilsError("Tests failed")
+from setuptools import Extension, setup
+from setuptools.command.build_ext import build_ext
+from setuptools.command.sdist import sdist
+from setuptools.errors import CCompilerError, ExecError, PlatformError
```

Comment:
Looks like `CCompilerError` can't be imported, not sure what's the new exception.

Reply:
setuptools vendored those exceptions from distutils. And it's only available on very recent releases. I'll revert this.